pytorch_optimizer/optimizer/schedulefree.py

```diff
@@ -428,6 +414,12 @@ def step(self, closure: CLOSURE = None) -> LOSS:
                 n_sma_threshold=4,
                 degenerated_to_sgd=group['degenerated_to_sgd'],
             )
+            if n_sma > 4:
+                # cancel bias correction2
+                lr = lr / bias_correction2_sq
+            elif lr < 0.:
+                # n_sma < 4.0 and degenerated_to_sgd is False
+                lr = 0.0
```

Comment:
it seems like your modification is different from the original ScheduleFreeRAdam optimizer [here](https://github.com/facebookresearch/schedule_free/blob/main/schedulefree/radam_schedulefree.py#L163). could you elaborate more on this?

as far as i know, usually in [`RAdam` optimizer](https://github.com/LiyuanLucasLiu/RAdam/blob/master/radam/radam.py#L156), parameter is not gonna updated in case of `lr` is negative (like your comment, n_sma is under the threshold and `degenerated_to_sgd` is `False`).

(of course, i mistakenly miss to handle that case in the current code), i think it'd be more efficient to skip the parameter update instead of setting the lr to 0.0!

Reply:
i get your point! well, I agree that even if the parameters aren't updated, setting lr to 0.0 or sth else seems great.